lib/sensor.js

```diff
@@ -223,6 +231,20 @@ function Sensor(opts) {
         return this.constrained;
       }
     },
+    freq: {
+      get: function() {
+        return priv.get(this).freq;
+      },
+      set: function(newFreq) {
+        priv.set(this, {
+          freq: newFreq
+        });
```

Comment:
Turns out that with the existing code, checking for that case is not easy.  The new freq get function is the only one that explicitly returns the private state value.  Everything else just uses the value as part of some calculation.  The *best* check I could come up with was to inject a value through analogRead, use booleanAt to set a barrier value that returns true for the value, then check that boolean was still true after setting the new freq property value.  Wiping out the booleanBarrier state value cased boolean to go from true to false.  Messy, and an incomplete check.

Is it *appropriate* to add a readonly state property to return the complete contents of the private data?  That could then be easily added to getShape() function I added for checking property changes.  Alternatively, and potentially more appropriate, defined property get functions could be created for each of the private state properties.  Something like: barrier, scale, val.  For my (testing) purposes, a read only state property would be easiest to use to provide full coverage.  The deepEqual check would automatically catch any unexpected changes.  New additions to the state properties would only need to be added to the defShape object to be fully included in all of the instance property validation done by the deepEqual tests .  That is actually what the shape checks were intended to catch.  Any unanticipated side effects.  But as is, it does not see changes in the instance private state properties.